### PR TITLE
Unify `tauchen` and `rouwenhorst` API

### LINKED ISSUE #363: Unify and improve `tauchen` and `rouwenhorst`
The arguments to these two functions have different order, even though they do something very similar.

I think it would be worth breaking backwards compatibility to fix this.  We just need to take care to update the lectures accordingly.  

Also, `ybar` is missing from `tauchen` and it would be good to have it.

See http://quanteconpy.readthedocs.io/en/latest/markov/approximation.html

I propose that we change the signature of `tauchen` to `tauchen(n, ybar, sigma, rho, m=3)`.

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #363 
#663

### CODE REVIEW

**Comment by oyamad** [comment]
Maybe add a note after this?

https://github.com/QuantEcon/QuantEcon.py/blob/e1c08e0e740c6e370ca9100901802e80d5eed93b/quantecon/markov/approximation.py#L25-L55

Something like "where `p = q = (1 + rho) / 2`."

**Review by oyamad — APPROVED** [review]
It looks good to me!

**Comment by jstac** [comment]
Nice work @Smit-create !  

This looks good to me too but please coordinate with @mmcky to make sure the lectures are suitably modified.

**Comment by Smit-create** [comment]
Thanks for the review. Once this is merged, I will quickly start fixing the lecture codes.

**Comment by mmcky** [comment]
> Thanks for the review. Once this is merged, I will quickly start fixing the lecture codes.

@Smit-create the best way to go here is to install this branch version on a PR marked `do not merge` and you can use that for testing. You just need to update the `environment.yml` file to use the branch version using `git+https`. I will try and set that up for you later this evening if not sure of the syntax.

File: quantecon/markov/approximation.py
Comment (mmcky) [inline quantecon/markov/approximation.py:15]: @Smit-create while it is difficult in this change to preserve backward compatibility, can you add a `deprecation` notice to the doctoring at the bottom.

Comment (Smit-create) [reply quantecon/markov/approximation.py:16]: Done, thanks!

**Comment by mmcky** [comment]
@Smit-create I will merge this into `master` but we will need lecture PR's tested prior to releasing `v0.6.0`

**Pull request merged**
